src/storage/src/source/kafka.rs

```diff
@@ -657,23 +656,18 @@ fn construct_source_message(
         ),
         _ => None,
     };
-    Ok(SourceMessage {
+    let pid = PartitionId::Kafka(msg.partition());
+    let Ok(offset) = u64::try_from(msg.offset()) else {
+        panic!("got negative offset ({}) from otherwise non-error'd kafka message", msg.offset());
```

Comment:
High-speed let-else adoption here!

I'm not 100% on panicking here: while this is a truly exceptional case, it's not one that indicates a bug in `materialize`. Maybe this should be a `halt`, if we don't think it's wise to percolate it upwards?

Reply:
hm, I think `halt` is meant to be used when the process stops because we know it should, for example if we know that we got fenced off. Honestly I'm not sure why `rdkafka` exposes the wrong data type here but putting an error result up the chain just for this exceptional case felt an overkill which is why I put the panic

Reply:
I think it inherits this from the JVM, where it's canonically a `long` (ie. `i64`).

Anyways, not that important, fine to leave it if you disagree.